forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,177 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import importlib
+from typing import Iterator, Tuple
+from collections import OrderedDict
+from forte.common import Resources
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""
+    A generic dataset reader class for classification datasets.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.index2class = None
+        self.label2index = None
+
+    def set_up(self):
+        assert self.configs.index2class is not None, (
+            "User must set index2class to enable"
+            " the dataset reader encode labels correctly"
+        )
+
+        # class and index
+        if not self.configs.digit_label:
+            # initialize class
+            self.index2class = self.configs.index2class
+            self.label2index = {
+                v: k for k, v in self.index2class.items()
+            }
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+
+    def _collect(self, csv_file: str) -> Iterator[Tuple[int, str]]:
+        with open(csv_file, encoding="utf-8") as f:
+            data = csv.reader(f, delimiter=",", quoting=csv.QUOTE_ALL)
+            next(data)
+            for line_id, line in enumerate(data):
+                yield line_id, line
+
+    def _cache_key_function(self, line_info: Tuple[int, str]) -> str:
+        return str(line_info[0])
+
+    def _parse_pack(self, line_info: Tuple[int, str]) -> Iterator[DataPack]:
+        line_id, line = line_info
+        # content label title
+        pack = DataPack()
+
+        # subtext fields must follow the ontology names
+        data_fields = self.configs.data_fields
+        assert (
+            "label" in data_fields
+        ), "There must be label data field in reader config."
+        assert len(self.configs.data_fields) == len(line), (
+            "Data fields provided in config "
+            "is not aligned with the actual line info from dataset."
+            + str((data_fields, line))
+        )
+        assert (
+            len(self.configs.subtext_fields) > 0
+        ), "There must be at least one subtext to reader to select from"
+
+        df_dict = OrderedDict()
+        for df, value in zip(data_fields, line):
+            df_dict[
+                df
+            ] = value  # in general, value can be subtext or other types of data
+
+        # it determines the order of concatenation
```

Comment:
Such information should be known to the user, not just as inline comment.

Reply:
Also mentioned it in default configs https://github.com/feipenghe/forte/blob/feaa1f3c6afcba2417e108caab29bdeb34388a72/forte/data/readers/classification_reader.py#L189